hledger-mode.el

```diff
@@ -125,15 +130,40 @@ COMMAND, ARG and IGNORED the regular meanings."
 (defun hledger-font-lock-keywords-1 ()
   "Minimal highlighting expressions for hledger mode."
   (list
-   `(,(concat "^  +" hledger-account-regex) 1 hledger-account-face)
-   `(,(concat "^account " hledger-account-regex "$") 1 hledger-account-face)
-   `(,(concat "^alias \\([^[:space:]=;\n]+\\) = " hledger-account-regex "$")
+   `(,(concat "^  +" hledger-account-regex) (1 hledger-account-face)
```

Comment:
What do you think about providing a function that a user can add to `hledger-mode-hook` if they would like ":" within account strings to be highlighted? We can probably used `font-lock-add-keywords` in the definition of such a function. That would keep the default code minimal and users who would like to have this feature can enable it.

Reply:
I like the idea of avoiding the extra work unless requested, but I’m not sure about using a function, since it would need a way to modify the list of rules returned by `hledger-font-lock-keywords-1`. I don’t think we could do it without adding a layer of indirection somewhere (so, for example, the user could set `hledger-font-lock-function` to `hledger-font-lock-with-colons` or something similar).

Alternatively, there could be a setting named something like `hledger-highlight-account-colons` (`nil` by default) that controls whether to insert rules for the colons into `hledger-font-lock-keywords-1`’s return value, since the list is recreated every time anyway.

Reply:
What do you think about just providing a single function that modifies font-lock rules with `font-lock-add-keywords`? Each of the added keyword rules can have the `override` flag turned on. This function let's say: `hledger-font-lock-highlight-colons`, can then be added by a user to their `hledger-mode-hook` and then will override the rules set up by `hledger-mode`.

Reply:
I think very few users would want this feature to be implemented. For features like this that are useful for me but not general enough, I usually add them to a personal version of `hledger-mode`: https://github.com/narendraj9/dot-emacs/tree/master/packages/rest/hledger-mode :-D

Reply:
I’m alright with leaving it out of this repository. I opened the PR just in case, but I wasn’t expecting it to be widely useful. I’ll close this and keep the customizations in my own configuration. Thank you for the discussion.

> In the long run, using [a tree-sitter grammar](https://github.com/cbarrete/tree-sitter-ledger) and parsing-based font-locking is going to be the better option.

I had the same thought! I was musing on what an hledger grammar would look like yesterday. (Even there, though, I have to imagine highlighting colons would be the very definition of a niche feature.)

Reply:
Thank you for taking the time though :) It gave me an opportunity to learn more about `font-lock-keywords`. :)

Reply:
Not at all. I’d never used an anchored highlighter before, so I learned more too.